examples/tide/src/main.rs

```diff
@@ -1,24 +1,106 @@
-// And finally, include the generated code for templates and static files.
-include!(concat!(env!("OUT_DIR"), "/templates.rs"));
-
+//! An example of how ructe can be used with the tide framework.
 mod ructe_tide;
 use ructe_tide::Render;
 
-use tide::{Response, StatusCode};
+use httpdate::fmt_http_date;
+use std::future::Future;
+use std::io::{self, Write};
+use std::pin::Pin;
+use std::str::FromStr;
+use std::time::{Duration, SystemTime};
+use templates::statics::{cloud_svg, StaticFile};
+use tide::http::headers::EXPIRES;
+use tide::http::{mime, Error};
+use tide::{Next, Redirect, Request, Response, StatusCode};
 
+/// Main entry point.
+///
+/// Set up an app and start listening for requests.
 #[async_std::main]
 async fn main() -> Result<(), std::io::Error> {
     let mut app = tide::new();
-
-    app.at("/").get(|_| async {
-        let mut res = Response::new(StatusCode::Ok);
-        res.render_html(|o| Ok(templates::hello(o, "world")?))?;
-        Ok(res)
-    });
+    app.middleware(handle_error);
+    app.at("/static/*path").get(static_file);
+    app.at("/favicon.ico")
+        .get(Redirect::new(format!("/static/{}", cloud_svg.name)));
+    app.at("/").get(frontpage);
 
     let addr = "127.0.0.1:3000";
     println!("Starting server on http://{}/", addr);
     app.listen(addr).await?;
 
     Ok(())
 }
+
+/// Handler for a page in the web site.
+async fn frontpage(_req: Request<()>) -> Result<Response, Error> {
+    // A real site would probably have some business logic here.
+    let mut res = Response::new(StatusCode::Ok);
```

Comment:
Would be good to include one example for the builder pattern and one without it.

Reply:
Thanks for the review!

> Latest tide contains the builder pattern so would be good to include for those too. I see to be using builder pattern with render more. Here is the code feel free to copy paste.

Ah, it's new?  So that's why you didn't use it in the code you contributed, I was wondering. Then I'll change the code to use it.

Reply:
Done.